xapian-core/cluster/cluster.cc

```diff
@@ -121,10 +85,11 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(MSetDocumentSource docs) {
+TermListGroup::add_documents(const MSet &docs) {
     LOGCALL_VOID(API, "TermListGroup::add_documents()", docs);
-    while (!docs.at_end())
-	add_document(docs.next_document());
+    MSetIterator it = docs.begin();
```

Comment:
Yes will do that. I seem to initialize iterators in a similar way in other parts of the code though. Would I need to change that too? Or was this only in relation to MSet because its part of Xapian's public API?